proxy/upgrade.go

```diff
@@ -161,7 +161,7 @@ func (p *upgradeProxy) serveHTTP(w http.ResponseWriter, req *http.Request) {
 	done := make(chan struct{}, 2)
 
 	if p.useAuditLog {
-		copyAsync("backend->request+audit", backendConn, io.MultiWriter(requestHijackedConn, p.auditLogOut), done)
+		copyAsync("backend->request+audit", backendConn, io.MultiWriter(p.auditLogOut, requestHijackedConn), done)
```

Comment:
Does this help or is this unrelated?

Reply:
It is the fix - this way audit log is always written first. I think it also makes sense to have it in the audit log even if subsequent write to connection fails.